bears/general/URLBear.py

```diff
@@ -105,7 +105,22 @@ def extract_links_from_file(file, link_ignore_regex, link_ignore_list):
             https?://                   # http:// or https:// as only these
                                         # are supported by the ``requests``
                                         # library
-            [^.:%\s_/?#[\]@\\]+         # Initial part of domain
```

Comment:
you could just add () to this here.  Would that not work?

Reply:
The task was not to make a group but correct the regex. The previous implementation allowed invalid keywords as I have already mentioned in my commit message.

Reply:
if you add `()` to this rule then your test case would pass. You should add more test cases where this would fail even with `()`.

Reply:
Yup, that can be done.